error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ImmutableSetRules.java

```diff
@@ -211,4 +214,43 @@ ImmutableSet<T> after(T e1, T e2, T e3, T e4, T e5) {
       return ImmutableSet.of(e1, e2, e3, e4, e5);
     }
   }
+
+  /** Prefer {@link Sets#intersection(Set, Set)} over more contrived alternatives. */
+  static final class SetsIntersection<T> {
+    @BeforeTemplate
+    ImmutableSet<T> before(Set<T> set1, Set<T> set2) {
+      return set1.stream().filter(set2::contains).collect(toImmutableSet());
+    }
+
+    @AfterTemplate
+    ImmutableSet<T> after(Set<T> set1, Set<T> set2) {
+      return Sets.intersection(set1, set2).immutableCopy();
```

Comment:
Next to `intersection` there's also `difference`. (And `union`; see the internal Store PR I filed a few days ago.)

Reply:
For reference: [#2545](https://github.com/PicnicSupermarket/picnic-store/pull/2524/files)

Reply:
@mohamedsamehsalah we try not to link to private things in this repo; I left it out on purpose. :upside_down_face: